### DISCUSSION TITLE
Run Camera Unbridged

### DISCUSSION THREAD

**Discussion by bjornand1** [Sunoo/homebridge-camera-ffmpeg#1067] [Q&A] [answered]
Hi, I just noticed the recommended option to run the cameras unbridged.

I do have issues with long update times in HomeKit, but I believe this is related to my IKEA Trådfri devices, while my HomeBridge system seems snappy. Is there any reason to change the settings for my two cameras? And if so, is there any risk of creating problems in doing so? Or is it normally easy to re-add the cameras after unbridging them?

**Comment by Sunoo** [comment, marked as answer]
If you enable debug, your Homebridge log will show how long it takes to pull a snapshot from your camera.

Keep in mind that HomeKit can only talk to one device on the bridge at a time, so any time spent waiting on a snapshot (or any other attributes from any other plugins, video streams work differently though) is a time where everything else on the bridge will not be able to respond.

If is certainly possible your cameras respond fast enough to not be worth worrying about unbridge, but that’s a call you’d have to make for yourself.

**Reply by bjornand1** [reply]
Thanks, then I think I'll just leave it as it is, in fear of breaking something works well enough.

**Reply by bjornand1** [reply]
By the way, is there any reason to fear that I will have problems re-adding the cameras if I check the unbridged box?

**Reply by Sunoo** [reply]
Some people have reported issues that required old cached accessories to be removed (I’m looking into a better way to handle that), but worst case you can always flip back to bridge mode.

**Reply by bjornand1** [reply]
OK, thanks!

**Reply by bjornand1** [reply]
@Sunoo Thanks for your very quick reply! So I tried unbridging it, but I am not sure which HomeKit code to use when adding the camera again?

**Reply by Sunoo** [reply]
Same as your main Homebridge code.

**Reply by bjornand1** [reply]
Using that code, it comes up as a bridge and the message that this accessory is already added...

**Reply by bjornand1** [reply]
Probably what you said about removing old cached accessories? Is there a way to do that without risking problems with the devices that should remain?

**Reply by Sunoo** [reply]
Are you running Config UI X?

**Reply by bjornand1** [reply]
Yes

**Reply by Sunoo** [reply]
You can remove the stale devices under Homebridge Settings -> Unpair Bridges / Cameras. After removing them, restart Homebridge and you should be able to pair the cameras.

**Reply by bjornand1** [reply]
@Sunoo Thanks, I tried that, but for some reason the camera is there after restarting. I have checked the "Unbridged" box for the camera and deleted it, but it is still there. Is there any other way to truly delete the camera?

**Reply by bjornand1** [reply]
I see various recipes in how to remove outdated devices, e.g.:

Delete persist folder
Delete cachedAccessories files
Restart homebridge

I am wary of going down this route out of fear that I will mess up my setup. My various Homebridge devices are part of lots of HomeKit automations and it would be much work to recreate these if I mess up the devices I want to keep.

So I guess my question is; is this a safe way to get my two cameras that I want to run unbridged purged from Homebridge?

**Reply by Sunoo** [reply]
If you delete all those folders, it will reset all devices on you bridge.

Removing them from Config UI where I said and restarting Homebridge should be enough to make them appear when you try to pair.

**Reply by bjornand1** [reply]
Thanks, I have tried that multiple times, but both cameras reappear under the purge screen in the Config UI, and thus cannot be added as separate devices in HomeKit. Don’t really know how to proceed, but I definitely don’t want to reset my devices!

**Reply by Sunoo** [reply]
They should reappear on that screen on reboot, since my plugin will set them up again when unbridge is set.

**Reply by bjornand1** [reply]
I see, that makes sense. But I wonder why I get the error message that the device has already been added when I try to add unbridged cameras to HomeKit?

**Reply by bjornand1** [reply]
Just to be absolutely sure, after unbridging a camera and restarting Homebridge, when I then proceed to add the camera as a device in HomeKit, I am supposed to use the HomeKit code of the Homebridge server? But if I unbridge both the cameras in question, how would HomeKit be able to distinguish between them?

**Reply by Sunoo** [reply]
You should see multiple devices when using that code.

**Reply by bjornand1** [reply]
Just not figuring this out, but I truly appreciate your trying to help out! I attach some screenshots, when I try to add a new device using the Homebridge code, I see only one device and that is a bridge.
![971F570A-E51D-42A5-9970-8D1A77C32A74](https://user-images.githubusercontent.com/36516146/110853780-52371d80-82b4-11eb-81da-52fff0f00b31.jpeg)
![C4EBC80C-54B8-4FD8-9EF2-0D4F36CCB530](https://user-images.githubusercontent.com/36516146/110853788-5400e100-82b4-11eb-981d-0a25104cc3ef.jpeg)
![6A492EDF-B547-4A8B-9760-A090FED3E81B](https://user-images.githubusercontent.com/36516146/110853790-54997780-82b4-11eb-98a3-e800029b5271.png)
![2857B1AA-F065-44F5-9937-519DB00716F9](https://user-images.githubusercontent.com/36516146/110853802-55caa480-82b4-11eb-99dd-89e0ea9f91f2.jpeg)

**Reply by bjornand1** [reply]
OK, no idea why, don't think I did anything differently, but all of a sudden it works!

**Reply by Sunoo** [reply]
Fantastic, glad you worked it out. Sorry I’ve been slow to respond at times, life’s been crazy.

**Reply by bjornand1** [reply]
Absolutely nothing to be sorry for; truly appreciate your help and maintaining the plugin!

**Reply by jwpereira2112** [reply]
I'm running Hoobs and I don't have the checkbox. Is there an example of how to add this to the config? Thanks!!!

**Reply by Sunoo** [reply]
> I'm running Hoobs and I don't have the checkbox. Is there an example of how to add this to the config? Thanks!!!

Set unbridge to true at the same level as the camera name.

**Reply by jwpereira2112** [reply]
That worked. I had to reset hoobs, and remove then re add Homebridge to Homekit. Then I could add secondary devices in Homekit buy saying I did not have a code, and then the un-bridged cameras appeared to add. Thanks!

**Reply by Sunoo** [reply]
Great. I’m not as familiar with HOOBS still, but I did get a setup for it shortly before my move. Once I find the box that got packed in, I’ll be testing future versions on HOOBS as well as Homebridge, and getting myself up to speed on their system.
